wgpu/tests/common/mod.rs

```diff
@@ -178,8 +170,11 @@ impl TestParameters {
         self
     }
 }
-
-pub fn initialize_test(parameters: TestParameters, test_function: impl FnOnce(TestingContext)) {
+pub fn initialize_test(
+    parameters: TestParameters,
+    limits_function: impl FnOnce(&Adapter) -> wgpu::Limits,
```

Comment:
Why is this needed, the test should require the very minimum limits it requires and behave accordingly. If there are multiple "modes" to test, there should be multiple tests.